File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/StoredRoutingTask.java
package net.cyclestreets.routing;

import net.cyclestreets.view.R;
import net.cyclestreets.content.RouteData;
import net.cyclestreets.content.RouteDatabase;

import android.content.Context;

public class StoredRoutingTask extends RoutingTask<Integer>
{
	private final RouteDatabase db_;

	StoredRoutingTask(final RouteDatabase db,
	                  final Context context)
	{
		super(R.string.route_loading, context);
		db_ = db;
	} // StoredRoutingTask

	@Override
	protected RouteData doInBackground(Integer... params)
	{
		return db_.route(params[0]);
	} // doInBackground
} // StoredRoutingTask

File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/StoredRoutingTask.kt
package net.cyclestreets.routing

import android.content.Context

import net.cyclestreets.content.RouteData
import net.cyclestreets.content.RouteDatabase
import net.cyclestreets.view.R

internal class StoredRoutingTask (private val db: RouteDatabase,
                                  context: Context) : RoutingTask<Int>(R.string.route_loading, context) {
    override fun doInBackground(vararg params: Int?): RouteData {
        return db.route(params[0]!!)
    }
}
